sdx-encrypted-submitter.go

```diff
@@ -56,28 +73,40 @@ func main() {
 	marshalError := yaml.Unmarshal(yamlFile, &config)
 	exitOnError(marshalError, fmt.Sprintf("unable to unMarshal yaml from %s", configFileName))
 
-	message, messageError := getMessage(config.MessageFilePath)
+	message, messageError := getRawMessage(config.MessageFilePath)
 	exitOnError(messageError, "could not read message body")
 
-	// If encrypt specified then encrypt
-	// if sign specified then sign
+	var mappedData map[string]interface{}
+	fileErr := json.Unmarshal(message, &mappedData) //file contents are arbitrary Json
+	exitOnError(fileErr, "Could not marshal Json from input file")
+
+	txID = fmt.Sprintf("%v", mappedData["tx_id"])
+	if txID == "<nil>"{
+		exitOnError(errors.New("No 'tx_id' field in input Json"), "Processing halted")
+	}
+
+	jwe, tokenError := authentication.GetJwe(mappedData, config.SigningKeyFile, config.EncryptionKeyFile)
+	message = []byte(jwe)
+	if tokenError != nil {
+		exitOnError(errors.New("Processing halted"), fmt.Sprintf("%s %s", (*tokenError).From, (*tokenError).Desc))
+	}
 
 	url := fmt.Sprintf("amqp://%s:%s@%s:%d/%s", config.Name, config.Password, config.Host, config.Port, config.Vhost)
-	rabbitError := sendToRabbit(url, config.Exchange, config.RoutingKey, message)
+	rabbitError := sendToRabbit(url, config.Exchange, config.RoutingKey, txID, message)
 	exitOnError(rabbitError, "unable to send message to rabbitmq")
 
 	fmt.Printf("message from file:'%s' published to exchange:'%s' using routing key:'%s\n", config.MessageFilePath, config.Exchange, config.RoutingKey)
 }
 
 func exitOnError(err error, msg string) {
 	if err != nil {
-		fmt.Println("%s: %s", msg, err)
+		fmt.Println(msg, " - ", err)
```

Comment:
If I remember correctly log.Fatal will print the message and exit the process, so this function is not needed

Reply:
There is no logging in the app as its a command line utility.